apps/04TowersOfHanoi.js

```diff
@@ -13,34 +13,62 @@ var stacks = {
   c: []
 };
 
+// resets stacks
+function startOver() {
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  }
+};
+
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // Pops from start stack and then pushes onto end stack
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
+  var endPieceFrom = stacks[startStack][stacks[startStack].length - 1];
+  var endPieceTo = stacks[endStack][stacks[endStack].length - 1];
+  if (endPieceFrom < endPieceTo || endPieceTo === undefined) {
+    movePiece(startStack, endStack);
+    return true
+  } else if (endPieceFrom > endPieceTo) {
+    console.log("Invalid Move");
+    return false
+  }
 
 }
 
 function checkForWin() {
   // Your code here
+  if (stacks.c.length === 4 || stacks.b.length === 4) {
+    console.log("Congratulations, You Won!!");
+    startOver();
+    return true;
+    // process.exit();
+  } else {
+    return false;
+  }
 
 }
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
-
+  isLegal(startStack, endStack);
 }
 
 function getPrompt() {
   printStacks();
+  checkForWin();
```

Comment:
This is OK here, but it'd also probably fit better in the `towersOfHanoi` function